### PR TITLE
Add practical session section

### PR DESCRIPTION
**Pull request by RazvanN7**

### CODE REVIEW

File: description.md [hunk 1]
```diff
@@ -34,6 +34,50 @@ Put differently, it shows how each content type is to be used in the various tea
...
+- one or more quizzes regarding some aspects of what has been presented so far.
+- one or more drills where the learner applies the presented concept.
+
+There is no enforcement on the order in which the above content types should be present in the concept layout, because that depends on the nature of the presented concept.
+However, we do recommend that the content types are mingled toghether so that the learner's attention is kept alive.
+For example, having a large theoretical description or a large demo may bore some students.
+To avoid that, quizzes, drills and tutorials may be sprinkled throught the concept layout to assure more engagement.
+
+Ideally, each practical activity (quiz, tutorial, drill) would be automatically evaluated and feedback would be instantly generated, if possible.
+For example, in computer programming courses, where drills and tutorials are typically comprised of writting programs, a checker should also accompany them.
```
Comment (gabrielmocanu) [inline description.md:69]:
```suggestion
For example, in computer programming courses, where drills and tutorials are typically comprised of writing programs, a checker should also accompany them.
```

File: description.md [hunk 2]
```diff
@@ -34,6 +34,50 @@ Put differently, it shows how each content type is to be used in the various tea
 
 ### Practical Sessions
 
+Practical sessions are represented by hands-on activities that learners perform in order to acquire practical experience.
```
Comment (gabrielmocanu) [inline description.md:37]:
```suggestion
Practical sessions are represented by hands-on activities that learners perform in order to acquire experience.
```

File: description.md [hunk 3]
```diff
@@ -34,6 +34,50 @@ Put differently, it shows how each content type is to be used in the various tea
...
+These activities are typically supervized by a teacher that proposes the practical exercices and evaluates the learner's performance.
+The purpose of the evaluation is mainly to provide constructive feedback, as opposed to grading the student.
```
Comment (gabrielmocanu) [inline description.md:39]:
```suggestion
The purpose of the evaluation is mainly to provide constructive feedback and support, as opposed to grading the student.
```

**Review by gabrielmocanu — CHANGES_REQUESTED** [review]

File: description.md [hunk 4]
```diff
@@ -34,6 +34,50 @@ Put differently, it shows how each content type is to be used in the various tea
...
+
+Instances of practical sessions are represented by laboratories, seminars or workshops.
+The format of a practical session may include:
+
+- theoretical and practical explanations offered by the teacher.
```
Comment (Alex-deVis) [inline description.md:44]:
```suggestion
- theoretical explanations and practical examples offered by the teacher.
```

File: description.md [hunk 5]
```diff
@@ -34,6 +34,50 @@ Put differently, it shows how each content type is to be used in the various tea
...
+- practical activities performed by the learners.
+- debates and discussions between students and teacher or between students themselves regarding specific topics.
+- feedback from the teacher or peers.
+
+The presence and length of each of these activities during a practical session is influenced by the preference of the teacher.
```
Comment (Alex-deVis) [inline description.md:49]:
```suggestion
The presence and length of each of these activities during a practical session are a matter of the teacher's preference.
```

File: description.md [hunk 6]
```diff
@@ -34,6 +34,50 @@ Put differently, it shows how each content type is to be used in the various tea
...
+As such, the educational materials used as support for the practical sessions should offer the flexibility of being tailored depending on the teacher's needs.
+At the same time, they should be usable in the absence of a teacher as standalone learning materials for self-taught individuals that want to learn by themselves.
+
+To meet these constraints, educational materials for practical sessions should abide to the following format.
+A practical sessions is comprised of a series of concepts that the learner must assimilate.
```
Comment (Alex-deVis) [inline description.md:54]: Both sentences serve as an introduction, I think it is clearer with only the second one.

File: description.md [hunk 7]
```diff
@@ -34,6 +34,50 @@ Put differently, it shows how each content type is to be used in the various tea
...
+A concept must contain:
+
+- a brief theoretical description (which may contain links to more broader explanations) by using reading and media materials.
+- one or more practical exemplifications of the described concept by using demos.
+- one or more tutorials where the learner makes the first steps in applying the described concept.
+- one or more quizzes regarding some aspects of what has been presented so far.
+- one or more drills where the learner applies the presented concept.
+
+There is no enforcement on the order in which the above content types should be present in the concept layout, because that depends on the nature of the presented concept.
```
Comment (Alex-deVis) [inline description.md:63]:
```suggestion
There is no enforcement on the order in which the above content types should appear in the concept layout because that depends on the nature of the presented concept.
```

File: description.md [hunk 8]
```diff
@@ -34,6 +34,50 @@ Put differently, it shows how each content type is to be used in the various tea
...
+However, we do recommend that the content types are mingled toghether so that the learner's attention is kept alive.
+For example, having a large theoretical description or a large demo may bore some students.
+To avoid that, quizzes, drills and tutorials may be sprinkled throught the concept layout to assure more engagement.
+
+Ideally, each practical activity (quiz, tutorial, drill) would be automatically evaluated and feedback would be instantly generated, if possible.
```
Comment (Alex-deVis) [inline description.md:68]:
```suggestion
Ideally, each practical activity (quiz, tutorial, drill) would be automatically evaluated and feedback would be instantly generated.
```
"Ideally" and "if possible" mean the same thing here.

**Review by Alex-deVis — APPROVED** [review]
The idea is well presented, I will adapt the assignments chapter to match.
I left suggestions inline.

File: description.md (hunk 3 above)
Comment (razvand) [reply description.md:39]: I would refrain from using the word "constructive". It doesn't actually say anything.

Also, I wouldn't say the primary purpose is feedback. The primary purpose is to grain practical skills, to learn by doing.

File: description.md [hunk 9]
```diff
@@ -34,6 +34,50 @@ Put differently, it shows how each content type is to be used in the various tea
 
 ### Practical Sessions
 
+Practical sessions are represented by hands-on activities that learners perform in order to acquire practical experience.
+These activities are typically supervized by a teacher that proposes the practical exercices and evaluates the learner's performance.
+The purpose of the evaluation is mainly to provide constructive feedback, as opposed to grading the student.
+
+Instances of practical sessions are represented by laboratories, seminars or workshops.
```
Comment (razvand) [inline description.md:41]:
```suggestion
Instances of practical sessions are represented by lab sessions, seminars or workshops.
```

I wouldn't use "laboratories", since that term is usually reserved for physical rooms.

File: description.md (hunk 4 above)
Comment (razvand) [inline description.md:44]: The first item should be "support material" in the form of reading content and media, and slides.

File: description.md [hunk 10]
```diff
@@ -34,6 +34,50 @@ Put differently, it shows how each content type is to be used in the various tea
...
+The format of a practical session may include:
+
+- theoretical and practical explanations offered by the teacher.
+- practical activities performed by the learners.
+- debates and discussions between students and teacher or between students themselves regarding specific topics.
+- feedback from the teacher or peers.
```
Comment (razvand) [inline description.md:47]: For each bullet you should reference items in the section `Types of Content` that are used in practical sessions.

File: description.md (hunk 5 above)
Comment (razvand) [reply description.md:49]: I would reword it "Teacher preference, institutional recommendations and implementation specifics dictate:
* which types of practical sessions are used
* length of a practical session instance
* format of a practical session"

File: description.md [hunk 11]
```diff
@@ -34,6 +34,50 @@ Put differently, it shows how each content type is to be used in the various tea
...
+
+The presence and length of each of these activities during a practical session is influenced by the preference of the teacher.
+As such, the educational materials used as support for the practical sessions should offer the flexibility of being tailored depending on the teacher's needs.
+At the same time, they should be usable in the absence of a teacher as standalone learning materials for self-taught individuals that want to learn by themselves.
+
+To meet these constraints, educational materials for practical sessions should abide to the following format.
```
Comment (razvand) [inline description.md:53]: After "the following format" I would expect a listing, but that comes later. I would rephrase this.

File: description.md (hunk 6 above)
Comment (razvand) [inline description.md:54]: I would use the term "topics" instead of "concepts". "concept" is more theoretical, while "topic" sounds closer to the "learning unit".

File: description.md [hunk 12]
```diff
@@ -34,6 +34,50 @@ Put differently, it shows how each content type is to be used in the various tea
...
+A practical sessions is comprised of a series of concepts that the learner must assimilate.
+A concept must contain:
+
+- a brief theoretical description (which may contain links to more broader explanations) by using reading and media materials.
```
Comment (razvand) [inline description.md:57]: Do not use periods (`.`) at the end of bullet points.

File: description.md (hunk 1 above)
Comment (razvand) [inline description.md:69]:
```suggestion
For example, in Computer Programming courses, drills and tutorials typically require the student to write programs.
Checkers should accompany these activities to ensure automated verification.
```

File: description.md [hunk 13]
```diff
@@ -34,6 +34,50 @@ Put differently, it shows how each content type is to be used in the various tea
...
+However, we do recommend that the content types are mingled toghether so that the learner's attention is kept alive.
+For example, having a large theoretical description or a large demo may bore some students.
+To avoid that, quizzes, drills and tutorials may be sprinkled throught the concept layout to assure more engagement.
+
+Ideally, each practical activity (quiz, tutorial, drill) would be automatically evaluated and feedback would be instantly generated, if possible.
+For example, in computer programming courses, where drills and tutorials are typically comprised of writting programs, a checker should also accompany them.
+Quizzes, if employed, are automatically checked for correctness.
+Therefore, for humanitarian fields of study such as history or philosophy, we recommend using quizzes instead of drills as a form of practical activity.
+
+The number of concepts and the ordering in which the concepts are presented in a practical session is highly dependent on the nature of the field of study and the preference of the teacher.
```
Comment (razvand) [inline description.md:73]:
```suggestion
The number of topics and the ordering in which the topics are presented in a practical session is highly dependent on the nature of the field of study and the preference of the teacher.
```

File: description.md [hunk 14]
```diff
@@ -34,6 +34,50 @@ Put differently, it shows how each content type is to be used in the various tea
...
+As a consequence, we do not have any recommendations regarding this aspect.
+However, if the content creator develops materials that have dependencies between them, this should be specified.
+
+Some teachers use slides to perform the brief theoretical description.
+Since this is a subjective choice, as content creators we do not provide slides for practical sessions.
+Rather, we create all of the concepts and it is up to the user to choose between them and create additional materials, such as slides.
```
Comment (razvand) [inline description.md:79]: This is imprecise. We do provide slides, and they can be used during practical activities. There is no reason the `slides` content type should only be used for lectures.

File: description.md (hunk 12 above)
[the remark this answers is not in the document]
Comment (razvand) [reply description.md:57]: and, potentially, slides

**Review by razvand — CHANGES_REQUESTED** [review]
This looks good. I think you covered most of the important topics. I made suggestions for clarifications and for improving wording.

File: description.md (hunk 3 above)
Comment (RazvanN7) [reply description.md:39]: The purpose of the evaluation is to provide feedback. The purpose of the practical session is to learn by doing as mentioned in the leading sentence.

Comment (RazvanN7) [reply description.md:39]: @gabrielmocanu also, the purpose of the evaluation is not to provide support, but to provide feedback.

File: description.md (hunk 4 above)
Comment (RazvanN7) [reply description.md]: No, this is not a description of the materials needed for practical sessions, but rather a description of the high-level activities that take place during a practical session. What you are asking for is treated below.

**Comment by RazvanN7** [comment]
@razvand @gabrielmocanu @Alex-deVis I have addressed your feedback.

File: description.md [hunk 15]
```diff
@@ -34,6 +34,52 @@ Put differently, it shows how each content type is to be used in the various tea
...
+- a brief theoretical description (which may contain links to broader explanations) by using [reading content](#reading), [media](#media), and, potentially, [slide](#slides) materials
+- one or more practical exemplifications of the described topic by using [demos](#demos)
+- one or more [tutorials](#tutorials) where the learner makes the first steps in applying the described topic
+- one or more [quizzes](#quizzes) regarding some aspects of what has been presented so far
+- one or more [drills](#drills) where the learner applies the presented topic
+
+There is no enforcement on the order in which the above content types should appear in the topic layout because that depends on the nature of the presented topic.
+However, we do recommend that the content types are mingled together so that the learner's attention is kept alive.
+For example, having a large theoretical description or a large demo may bore some students.
+To avoid that, quizzes, drills and tutorials may be sprinkled throught the topic layout to assure more engagement.
```
Comment (gabrielmocanu) [inline description.md:71]:
```suggestion
To avoid that, quizzes, drills and tutorials may be sprinkled throughout the topic layout to ensure more engagement.
```

File: description.md [hunk 16]
```diff
@@ -34,6 +34,52 @@ Put differently, it shows how each content type is to be used in the various tea
...
 ### Practical Sessions
 
+Practical sessions are represented by hands-on activities that learners perform in order to acquire experience.
+These activities are typically supervised by a teacher that proposes the practical exercises and evaluates the learner's performance.
+The purpose of the evaluation is mainly to provide feedback as opposed to grading the student.
+
+Instances of practical sessions are represented by lab sessions, seminars or workshops.
+The format of a practical session may include:
+
+- support material in the form of
```
[the remark this answers is not in the document]
Comment (RazvanN7) [reply description.md]: this was a slip. I started adding something, but then thought it's better this way.

**Review by gabrielmocanu — APPROVED** [review]
LGTM

File: description.md [hunk 17]
```diff
@@ -34,6 +34,51 @@ Put differently, it shows how each content type is to be used in the various tea
 
 ### Practical Sessions
 
+Practical sessions are represented by hands-on activities that learners perform in order to acquire experience.
+These activities are typically supervised by a teacher that proposes the practical exercises and evaluates the learner's performance.
```
Comment (Alex-deVis) [inline description.md:38]:
```suggestion
These activities are typically supervised by a teacher who proposes the practical exercises and evaluates the learner's performance.
```

File: description.md [hunk 18]
```diff
@@ -34,6 +34,51 @@ Put differently, it shows how each content type is to be used in the various tea
...
+The purpose of the evaluation is mainly to provide feedback as opposed to grading the student.
+
+Instances of practical sessions are represented by lab sessions, seminars or workshops.
```
Comment (Alex-deVis) [inline description.md:41]:
```suggestion
Instances of practical sessions are represented by lab sessions, seminars, or workshops.
```

File: description.md [hunk 19]
```diff
@@ -34,6 +34,51 @@ Put differently, it shows how each content type is to be used in the various tea
...
+The format of a practical session may include:
+
+- theoretical explanations and practical examples offered by the teacher
+- practical activities performed by the learners
+- debates and discussions between students and teacher or between students themselves regarding specific topics
+- feedback from the teacher or peers
+
+Teacher preference, institutional recommendations and implementation specifics dictate:
```
Comment (Alex-deVis) [inline description.md:49]: I like this sentence.

File: description.md [hunk 20]
```diff
@@ -34,6 +34,51 @@ Put differently, it shows how each content type is to be used in the various tea
...
+
+- which types of practical sessions are used
+- length of a practical session instance
+- format of a practical session
+
+As such, the educational materials used as support for the practical sessions should offer the flexibility of being tailored depending on the teacher's needs.
+At the same time, they should be usable in the absence of a teacher as standalone learning materials for self-taught individuals that want to learn by themselves.
+
+A practical session is comprised of a series of topics that the learner must assimilate.
```
Comment (Alex-deVis) [inline description.md:58]:
```suggestion
A practical session comprises a series of topics that the learner must assimilate.
```

**Review by Alex-deVis — APPROVED** [review]
Small touches.
